Integrations/python/test/test_learn.py

```diff
@@ -0,0 +1,304 @@
+# 
+# Copyright (c) 2016-2021 Deephaven Data Labs and Patent Pending
+# 
+
+##############################################################################
+# NOTE: the jvm should have been initialized, or this test will certainly fail
+##############################################################################
+
+import numpy as np
+import unittest
+import sys
+import os
+
+from deephaven import learn, tableToDataFrame, TableTools
+from deephaven.learn import gather
+
+class TestLearn(unittest.TestCase):
```

Comment:
I have some general comments on this Python test.  I'm making them as teaching comments to make better tests and code going forward.
1) The code is not very local, in the sense that data is spread far from where it is used -- variable scope is large.  It would be cleaner if the data was defined where it is used (e.g. in the test function), since I think it is only used in one place.
2) There is a ton of code duplication.  The test methods could be refactored to something like:
```
def base_test(...):
    ...
def test_byte():
    table = ...
    ...
    base_test(table, ...)
```
3) `learn` should not appear in these tests.  A unit test should be testing that a function is returning what it should.  By injecting `learn`, this is becoming more of an integration test... and also increasing complexity.
4) I assume this test should be called `test_learn_gather.py` or something, since that is what you are really trying to test, and not learn.  You could have a `test_learn.py` to just test `learn`, but it should be very simple and should not be testing `gather`.
5) There is transpose stuff in the unit tests.  Not sure why that is here, since transpose was removed.